pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -134,20 +140,44 @@ func loadVfioDrivers() {
 }
 
 func bindDeviceToVFIOPCIDriver(pd *v1beta1.PCIDevice) error {
-	logrus.Infof("Binding device %s vfio-pci", pd.Status.Address)
+	if deviceBoundToDriver(vfioPCIDriverPath, pd.Status.Address) {
+		return nil
+	}
+
+	vendorId := pd.Status.VendorId
+	deviceId := pd.Status.DeviceId
+	var id string = fmt.Sprintf("%s %s", vendorId, deviceId)
+	logrus.Infof("Binding device %s [%s] to vfio-pci", pd.Name, id)
+
+	newIdFile, err := os.OpenFile("/sys/bus/pci/drivers/vfio-pci/new_id", os.O_WRONLY, 0200)
+	if err != nil {
+		logrus.Errorf("Error opening new_id file")
+		return err
+	}
+	defer newIdFile.Close()
+	_, err = newIdFile.WriteString(id)
+	if err != nil && !os.IsExist(err) {
+		logrus.Errorf("Error writing to new_id file: %s", err)
+		return err
+	}
 
+	logrus.Infof("Binding device %s vfio-pci", pd.Status.Address)
 	file, err := os.OpenFile("/sys/bus/pci/drivers/vfio-pci/bind", os.O_WRONLY, 0200)
 	if err != nil {
-		logrus.Errorf("Error opening new_id file: %s", err)
+		logrus.Errorf("Error opening bind file: %s", err)
 		return err
 	}
 	_, err = file.WriteString(pd.Status.Address)
 	if err != nil {
-		logrus.Errorf("Error writing to new_id file: %s", err)
+		logrus.Errorf("Error writing to bind file: %s", err)
 		file.Close()
 		return err
 	}
 	file.Close()
+
+	if !deviceBoundToDriver(vfioPCIDriverPath, pd.Status.Address) {
+		return fmt.Errorf("no device %s found at /sys/bus/pci/drivers/vfio-pci", pd.Status.Address)
```

Comment:
use `bindDeviceToVFIOPCIDriver` instead.

Reply:
the error will result in object being requeued and falling through the reconcile loop